### DISCUSSION TITLE
UDM-IPTV won't start on UXG Pro

### DISCUSSION THREAD

**Discussion by CDRX2** [fabianishere/udm-iptv#284] [Q&A] [answered]
Hi,

I've just installed udm-iptv on my uxg pro in the hopes of getting it to work with my provider (Init7 in Switzerland). Although I followed the guide in [this discussion](https://github.com/fabianishere/udm-iptv/discussions/222) it seems my instance of udm-iptv won't start.

Here is the output of udm-iptv diagnose:
```=== Configuration ===
WAN Interface: eth2
WAN VLAN: 0 (dev iptv)
WAN DHCP: false (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 224.0.0.0/8  239.0.0.0/8  77.109.128.0/19
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Nov 01 16:44:00 Router udm-iptvd[34156]: DEBUG[init_interface@208]: if_mtu = 1500
Nov 01 16:44:00 Router udm-iptvd[34156]: DEBUG[init_interface@214]: if_falgs = 4163
Nov 01 16:44:00 Router udm-iptvd[34156]: ERROR[init_interface@172]: SIOCGIFADDR filed eth2
Nov 01 16:44:00 Router udm-iptvd[34156]: ERROR[init_interface@232]: Disable IGMP, there is not igmp upstream interface or igmp downstream interface
Nov 01 16:44:00 Router udm-iptvd[34156]: FATAL[init_interface@251]: exiting.....
Nov 01 16:44:00 Router udm-iptvd[34156]: ERROR[main@619]: exiting.........
Nov 01 16:44:00 Router IMPROXY[34156]: FATAL[init_interface@251]: exiting.....
Nov 01 16:44:00 Router systemd[1]: udm-iptv.service: Main process exited, code=exited, status=1/FAILURE
Nov 01 16:44:00 Router IMPROXY[34156]: ERROR[main@619]: exiting.........
Nov 01 16:44:00 Router systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
```
From what I see, WAN and LAN are configured correctly (no VLANs whatsoever) but I don't understand why I get this error message regarding igmp upstream interface or igmp downstream interface.

Anybody have any ideas?

**Comment by fabianishere** [comment]
Can you check if `eth2` has an IP address?

**Comment by CDRX2** [comment]
It seems `eth2` does not. My WAN is actually labelled `ppp0` according to `ifconfig`. I suppose that means I need create a custom config file, correct?

**Reply by fabianishere** [reply, marked as answer]
Your WAN interface needs to be `ppp0` then. You can change it in `/etc/udm-iptv.conf`

**Reply by CDRX2** [reply]
Thanks, that worked. Still no streams, though, will do some more troubleshooting tomorrow and post another question once I have more information.
